torch2trt/torch2trt.py

```diff
@@ -625,6 +626,12 @@ def torch2trt(module,
     if output_names is None:
         output_names = default_output_names(len(outputs))
 
+    # Create optimization profiles. This allows for dynamic shapes, especially dynamic batch sizes.
+    opt_profile = builder.create_optimization_profile();
+    for input_name, input_shape in zip(input_names, input_shapes):
+        opt_profile.set_shape(input_name, (1,) + input_shape[1:], input_shape, input_shape)
```

Comment:
I should probably set the max here to be that of the provided `max_batch_size` (which may be greater than the provided input tensor)